components/functions-library-mgt/org.wso2.carbon.identity.functions.library.mgt/src/main/java/org/wso2/carbon/identity/functions/library/mgt/util/FunctionLibraryExceptionManagementUtil.java

```diff
@@ -0,0 +1,105 @@
+     * @param data  data to replace if message needs to be replaced.
+     * @return FunctionLibraryManagementClientException.
+     */
+    public static FunctionLibraryManagementClientException handleClientException(
+            FunctionLibraryManagementConstants.ErrorMessage error, String data) {
+
+        String message = includeData(error, data);
+        return new FunctionLibraryManagementClientException(error.getCode(), message);
+    }
+
+    public static FunctionLibraryManagementClientException handleClientException(
+            FunctionLibraryManagementConstants.ErrorMessage error, String data, Throwable e) {
+
+        String message = includeData(error, data);
+        return new FunctionLibraryManagementClientException(error.getCode(), message, e);
+    }
+
+    public static FunctionLibraryManagementClientException handleClientException(
+            FunctionLibraryManagementConstants.ErrorMessage error) {
+
+        String message = error.getMessage();
+        return new FunctionLibraryManagementClientException(error.getCode(), message);
+    }
+
+    /**
+     * This method can be used to generate a FunctionLibraryManagementServerException from
+     * FunctionLibraryManagementConstants.ErrorMessage object when no exception is thrown.
+     *
+     * @param error FunctionLibraryManagementConstants.ErrorMessage.
+     * @param data  data to replace if message needs to be replaced.
+     * @return FunctionLibraryManagementServerException.
+     */
+    public static FunctionLibraryManagementServerException handleServerException(
+            FunctionLibraryManagementConstants.ErrorMessage error, String data, Throwable e) {
+
+        String message = includeData(error, data);
+        return new FunctionLibraryManagementServerException(error.getCode(), message, e);
+    }
+
+    public static FunctionLibraryManagementServerException handleServerException(
+            FunctionLibraryManagementConstants.ErrorMessage error) {
+
+        String message = error.getMessage();
+        return new FunctionLibraryManagementServerException(error.getCode(), message);
+    }
+
+    public static FunctionLibraryManagementServerException handleServerException(
+            FunctionLibraryManagementConstants.ErrorMessage error, Throwable e) {
+
+        String message = error.getMessage();
+        return new FunctionLibraryManagementServerException(error.getCode(), message, e);
+    }
+
+    /**
+     * Include the data to the error message.
+     *
+     * @param error FunctionLibraryManagementConstants.ErrorMessage.
+     * @param data  data to replace if message needs to be replaced.
+     * @return message format with data.
+     */
+    private static String includeData(FunctionLibraryManagementConstants.ErrorMessage error, String data) {
+
+        String message;
+        if (StringUtils.isNotBlank(data)) {
+            message = String.format(error.getMessage(), data);
```

Comment:
This will be bit consuming and error prone. We assume the error should have exactly one string place holder which we can't enforce. Would be better to give the responsibility to the callee to format/build and pass the exact message.